pkg/controller/replication_controller.go

```diff
@@ -70,6 +72,8 @@ func (r RealPodControl) createReplica(namespace string, controller api.Replicati
 		desiredAnnotations[k] = v
 	}
 
+	desiredAnnotations[RCNAME_ANNOTATION] = controller.Name
```

Comment:
Keep in mind that the annotation will only be applied at creation time, so pods created via other means that are then "adopted" by this replication controller won't have the same annotation, and in the case that the replication controller is deleted and a new one created, the pods adopted by the new controller won't be annotated with the name of the new controller, either.

If you want dynamic resolution, that's a reverse label lookup.